internal/apiutil/mfserver/mfserver.go

```diff
@@ -0,0 +1,61 @@
+package mfserver
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/mainflux/mainflux/logger"
+)
+
+type Server interface {
+	Start() error
+	Stop() error
+}
+
+type BaseServer struct {
+	Ctx      context.Context
+	Cancel   context.CancelFunc
+	Name     string
+	Address  string
+	Port     string
+	CertFile string
+	KeyFile  string
+	Logger   logger.Logger
+	Protocol string
+}
+
+func stopAllServer(servers ...Server) error {
+	var err error
```

Comment:
What's this error for?

Reply:
To be able to aggregate all the errors. For instance, we have two errors out of 5 servers, we will be able to return all the errors rather than the error it checked first